### PR TITLE
Load dependencies and description from mod.conf if available

### PR DESCRIPTION
**Pull request by TeTpaAka**
Smaller version of https://github.com/minetest/minetest/pull/3392

Just load the existing information from a different source, no additional complexity added.

### CODE REVIEW

File: builtin/mainmenu/modmgr.lua [hunk 1]
```diff
@@ -286,21 +286,36 @@ end
...
-
-		local dependencyfile = io.open(filename,"r")
-
-		if dependencyfile then
-			local dependency = dependencyfile:read("*l")
-			while dependency do
+		local mod_conf = Settings(modfolder .. DIR_DELIM .. "mod.conf")
+		local dependencies = nil
+		if mod_conf then
+			dependencies = mod_conf:get("depends");
```
Comment (sfan5) [inline builtin/mainmenu/modmgr.lua]: Superfluous `;`

File: builtin/mainmenu/modmgr.lua [hunk 2]
```diff
@@ -286,21 +286,36 @@ end
...
+		end
+		if dependencies then
+			local list = dependencies:split()
```
Comment (sfan5) [inline builtin/mainmenu/modmgr.lua]: By what is split here? `" "`?

Comment (TeTpaAka) [inline builtin/mainmenu/modmgr.lua]: Split splits by default with ",", which should be the separator between mod names.

File: builtin/mainmenu/modmgr.lua (hunk 1 above)
Comment (TeTpaAka) [inline builtin/mainmenu/modmgr.lua]: Fixed

**Comment by paramat** [comment]
Sorry to question this again but, is it so bad to have seperate files 'depends.txt' 'mod.conf' and 'description.txt'? Many mods (all of mine) only have 'depends.txt'. Seperate files are more flexible to work with.
Should we not first discuss and decide whether it's a good idea to move towards having everything in 'mod.conf'?
I'm not opposed but also can't see much reason to do this.

File: builtin/mainmenu/modmgr.lua [hunk 3]
```diff
@@ -286,21 +286,36 @@ end
...
+			end
+		else
+			local filename = modfolder ..
+						DIR_DELIM .. "depends.txt"
+
+			local dependencyfile = io.open(filename,"r")
+
+			if dependencyfile then
+				local dependency = dependencyfile:read("*l")
+				while dependency do
```
Comment (kaeza) [inline builtin/mainmenu/modmgr.lua]: While you're changing things around here, can you change this to use `:lines()`?

File: builtin/mainmenu/tab_mods.lua [hunk 4]
```diff
@@ -68,17 +68,26 @@ local function get_formspec(tabview, name, tabdata)
...
+		local descriptiontext
+		if mod_conf then
+			descriptiontext = mod_conf:get("description")
+		end
+		if descriptiontext then
+			descriptionlines = core.splittext(descriptiontext, 42)
 		else
-			descriptionlines = {}
-			descriptionlines[#descriptionlines + 1] = fgettext("No mod description available")
+			error = nil
```
Comment (kaeza) [inline builtin/mainmenu/tab_mods.lua]: This can be safely removed.

File: builtin/mainmenu/modmgr.lua (hunk 3 above)
Comment (TeTpaAka) [inline builtin/mainmenu/modmgr.lua]: Done

**Pull request closed without merging**